build/tooling/Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM golang:1.22 AS builder
+
+RUN GO111MODULE=on go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.15.0
```

Comment:
I don't _think_ there's much benefit to this vs `go run sigs.k8s.io/controller-tools/cmd/controller-gen@v0.15.0 ...` Am I missing something?

Reply:
Sorry if I am missing something here but we just want the utility in the docker image before calling it in the makefile right?

I can certainly just run it in the Makefile if that simplifies things but I was hoping we can throw more things in this image eventually like (kustomize) if we still use it down the road.